### PR TITLE
ports gym wrapper to gymnasium

### PR DESCRIPTION
**Pull request by raghavauppuluri13**
for reference: https://gymnasium.farama.org/content/migration-guide/

### CODE REVIEW

File: robosuite/wrappers/gym_wrapper.py [hunk 1]
```diff
@@ -83,15 +84,20 @@ def _flatten_obs(self, obs_dict, verbose=False):
...
+    def reset(self, seed=None, options=None):
         """
-        Extends env reset method to return flattened observation instead of normal OrderedDict.
+        Extends env reset method to return flattened observation instead of normal OrderedDict and optionally resets seed
 
         Returns:
             np.array: Flattened environment observation space after reset occurs
         """
+        if seed is not None:
+            try:
```
Comment (amandlek) [inline robosuite/wrappers/gym_wrapper.py:95]: the try and except might be overkill - why not a simple `isinstance(seed, int)` check?

Comment (raghavauppuluri13) [reply robosuite/wrappers/gym_wrapper.py:95]: > the try and except might be overkill - why not a simple `isinstance(seed, int)` check?

That's fair! I can simplify it to that.

> also, should seed setting happen in the reset function? I'm not super familiar with the new gymnasium api, but I would think that the seed should be set before constructing the environment, or maybe in the environment init

In the new gymnasium api, they have seed passed in to the `env.reset`. From the docs: "Therefore, [reset()](https://gymnasium.farama.org/api/env/#gymnasium.Env.reset) should (in the typical use case) be called with a seed right after initialization and then never again."

File: robosuite/wrappers/gym_wrapper.py [hunk 2]
```diff
@@ -83,15 +84,20 @@ def _flatten_obs(self, obs_dict, verbose=False):
...
+        Extends env reset method to return flattened observation instead of normal OrderedDict and optionally resets seed
 
         Returns:
             np.array: Flattened environment observation space after reset occurs
         """
+        if seed is not None:
+            if isinstance(seed, int):
+                np.random.seed(seed)
+            else:
+                TypeError("Seed must be an integer type!")
```
Comment (amandlek) [inline robosuite/wrappers/gym_wrapper.py:98]: you need a `raise` statement here right? After that fix, I think we can merge

**Review by amandlek — COMMENTED** [review]
We can merge after a minor fix

File: robosuite/wrappers/gym_wrapper.py (hunk 2 above)
Comment (raghavauppuluri13) [reply robosuite/wrappers/gym_wrapper.py]: Sorry, missed that! Should be fixed now

**Review by yukezhu — APPROVED** [review]
LGTM

**Comment by yukezhu** [comment]
@raghavauppuluri13 thanks for your PR!

**Pull request merged**